kml/edit-kml-ground-overlay/src/main/java/com/esri/samples/edit_kml_ground_overlay/EditKMLGroundOverlaySample.java

```diff
@@ -104,30 +87,58 @@ public void start(Stage stage) {
       KmlDataset kmlDataset = new KmlDataset(kmlGroundOverlay);
 
-      // create a KML layer for the scene view
+      // create a KML layer using the KML dataset
       KmlLayer kmlLayer = new KmlLayer(kmlDataset);
 
-      // add the KML layer to the scene
+      // add the KML layer to the scene view
       sceneView.getArcGISScene().getOperationalLayers().add(kmlLayer);
 
-      // move the viewpoint to the ground overlay
+      // set the viewpoint to the ground overlay
       sceneView.setViewpoint(new Viewpoint(kmlGroundOverlay.getExtent(), new Camera(kmlGroundOverlay.getGeometry().getExtent().getCenter(), 1250, 45, 60, 0)));
 
+      // create a slider for adjusting the overlay opacity
+      Slider slider = new Slider(0, 1, 1);
+      slider.setMaxWidth(300);
+
       // add an event handler to the slider to apply opacity to the KML ground overlay based on the slider value
-      slider.setOnMouseReleased(event -> kmlGroundOverlay.setColor(ColorUtil.colorToArgb(new Color(0, 0, 0, slider.getValue()))));
-      slider.setOnMouseDragged(event -> kmlGroundOverlay.setColor(ColorUtil.colorToArgb(new Color(0, 0, 0, slider.getValue()))));
+      slider.setOnMouseReleased(event -> updateGroundOverlayOpacity(kmlGroundOverlay, slider.getValue()));
+      slider.setOnMouseDragged(event -> updateGroundOverlayOpacity(kmlGroundOverlay, slider.getValue()));
 
-      // add the map view and controls to the stack pane
-      stackPane.getChildren().addAll(sceneView, controlsHBox);
-      StackPane.setAlignment(controlsHBox, Pos.TOP_LEFT);
-      StackPane.setMargin(controlsHBox, new Insets(10, 0, 0, 10));
+      // create a controls box
+      VBox controlsVBox = new VBox();
+      controlsVBox.setBackground(new Background(new BackgroundFill(Paint.valueOf("rgba(0,0,0,0.5)"), CornerRadii.EMPTY,
+              Insets.EMPTY)));
+      controlsVBox.setPadding(new Insets(10.0));
+      controlsVBox.setSpacing(10);
+      controlsVBox.setMaxSize(250, 20);
+      controlsVBox.getStyleClass().add("panel-region");
+
+      // create a label for the slider
+      Label label = new Label("Overlay Opacity: ");
+      // add the slider and label to the controls box
+      controlsVBox.getChildren().addAll(label, slider);
+
+      // add the scene view and controls to the stack pane
+      stackPane.getChildren().addAll(sceneView, controlsVBox);
+      StackPane.setAlignment(controlsVBox, Pos.TOP_LEFT);
+      StackPane.setMargin(controlsVBox, new Insets(10, 0, 0, 10));
 
     } catch (Exception e) {
       // on any error, display the stack trace.
       e.printStackTrace();
     }
   }
 
+  /**
+   * Updates the color of a KmlGroundOverlay with a provided alpha value.
+   *
+   * @param kmlGroundOverlay a provided KmlGroundOverlay.
+   * @param alphaValue       a double value representing the opacity to be applied.
+   */
+  private void updateGroundOverlayOpacity(KmlGroundOverlay kmlGroundOverlay, double alphaValue) {
```

Comment:
I think we should make it even simpler with the following:

```
private void setGroundOverlayOpacity() {
    kmlGroundOverlay.setColor(ColorUtil.colorToArgb(new Color(0, 0, 0, slider.getValue())));
}
```

And then have in the start method:

```
   // add an event handler to the slider to apply opacity to the KML ground overlay based on the slider value
      slider.setOnMouseReleased(event -> setGroundOverlayOpacity());
      slider.setOnMouseDragged(event -> setGroundOverlayOpacity());
```